### PR TITLE
feat(tree): add type-ahead for tree

### PR DESCRIPTION
**Pull request by jiaoy14**
`Fixes #1, Fixes #2`

### CODE REVIEW

File: src/tree-view/utils.js [hunk 1]
```diff
@@ -189,7 +189,71 @@ export const toggleIsExpanded = (
...
+      nodeId,
+      chars,
+      closestOmmer,
+      getId,
+      false,
+      false,
+    )[0];
+  return foundid;
+};
+export const matchString = (
```
Comment (tajo) [inline src/tree-view/utils.js:220]: It would be great to write some unit tests for function

**Comment by tajo** [comment]
Vercel couldn't build this PR. Some [syntax error](https://vercel.com/uber-ui-platform/baseweb/mcun4vczg)?

File: src/tree-view/utils.js (hunk 1 above)
Comment (jiaoy14) [reply src/tree-view/utils.js:220]: I will change that function. I combined range search into DFS . As suggested by Chase (https://github.com/uber/baseweb/pull/3560#discussion_r463752404), I will change that to a direct search.

**Comment by jiaoy14** [comment]
> Vercel couldn't build this PR. Some [syntax error](https://vercel.com/uber-ui-platform/baseweb/mcun4vczg)?

yes, I may need some type check.

File: src/tree-view/utils.js (hunk 1 above)
Comment (chasestarr) [reply src/tree-view/utils.js:205]: The e2e test looks good. Can you also add a unit test for this function like here? https://github.com/uber/baseweb/blob/b16b9845f984ed62413e0069aef719ea4c1ab1b3/src/tree-view/__tests__/utils.test.js

**Comment by tajo** [comment]
It seems not to be working for some examples like:

https://baseweb-435iz1oqp.vercel.app/components/tree-view/#adding-custom-labels

**Comment by jiaoy14** [comment]
> It seems not to be working for some examples like:
> 
> https://baseweb-435iz1oqp.vercel.app/components/tree-view/#adding-custom-labels

Yes, for now it can't support customized label.

**Comment by chasestarr** [comment]
How could we best support customized labels? Combobox for example has a `mapOptionToString` prop so that we can run text filters on the list

**Comment by jiaoy14** [comment]
> How could we best support customized labels? Combobox for example has a `mapOptionToString` prop so that we can run text filters on the list

Emm, the existing structure gives developer much freedom for customizing.  

> How could we best support customized labels? Combobox for example has a `mapOptionToString` prop so that we can run text filters on the list

Now `tree` has a prop `getID` which is used for getting id from customized labels. It behaves similarly like `mapOptionToString`.  I can imagine we have another prop `getLabelName`  used for customized label. But I  am not sure if this is necessary because it needs developers to define it  and this only used for type-ahead when label is customized.

**Pull request merged**